tests/PasswordCheckerTest.php

```diff
@@ -4,79 +4,131 @@
 
 class PasswordCheckerTest extends TestCase
 {
-    const APP_SPECIFIC_OBVIOUS_PASSWORDS = ['appname', 'companyname', 'companyltd', 'robert', 'smith', 'bob@example.com'];
-    const USER_PASSWORD_HISTORY          = [
-        '$2y$11$V8Tvqr3nyMMQrR1xE/IFgepTahAprWKWfoh.xgN7ziWdwYRsZyzCu', // couldyouhearme2
-        '$2y$11$AFhINRcNKiKR/OKAb8Is8uxovaHMjHhpsVxP8rULLoJRkGi7hhXYa', // couldyouhearmeb4
-    ];
+    protected function setUp()
+    {
+        $this->checker = new PasswordChecker(['appname', 'companyname', 'companyltd', 'robert', 'smith', 'bob@example.com']);
+        $this->checker->setPreviousPasswords([
+            '$2y$11$V8Tvqr3nyMMQrR1xE/IFgepTahAprWKWfoh.xgN7ziWdwYRsZyzCu', // couldyouhearme2
+            '$2y$11$AFhINRcNKiKR/OKAb8Is8uxovaHMjHhpsVxP8rULLoJRkGi7hhXYa', // couldyouhearmeb4
+        ]);
+    }
+
+    public function test_valid_with_no_options()
+    {
+        $this->assertTrue((new PasswordChecker)->validate('canyouhearme1'));
+    }
 
     public function test_passes_due_to_new_password()
     {
-        $checker = new PasswordChecker(static::APP_SPECIFIC_OBVIOUS_PASSWORDS);
-        $this->assertTrue($checker->validate('canyouhearme1', static::USER_PASSWORD_HISTORY));
-        $this->assertNull($checker->getErrorMessage());
+        $this->assertTrue($this->checker->validate('canyouhearme1'));
+    }
+
+    /**
+     * @expectedException        FusionsPim\PhpPasswordChecker\PasswordException
+     * @expectedExceptionMessage New and confirmation passwords are different
+     */
+    public function test_fails_due_to_confirmation_mismatch()
+    {
+        $this->checker->setConfirmation('canyouhearme0');
+        $this->checker->validate('canyouhearme1');
     }
 
+    /**
+     * @expectedException        FusionsPim\PhpPasswordChecker\PasswordException
+     * @expectedExceptionMessage New password must be at least 10 characters long
+     */
     public function test_fails_due_to_short_password()
     {
-        $checker = new PasswordChecker(static::APP_SPECIFIC_OBVIOUS_PASSWORDS);
-        $this->assertFalse($checker->validate('abc', static::USER_PASSWORD_HISTORY));
-        $this->assertEquals('New password must be at least 10 characters long', $checker->getErrorMessage());
+        $this->checker->validate('abc');
```

Comment:
Is it worth adding a test using a multibyte string that would be >10 characters when using standard `strlen`?

Reply:
@WillJW done ta